wm/screens.go

```diff
@@ -72,105 +97,86 @@ func (xsp *x11ScreensProvider) ScreenForGeometry(x int, y int, width int, height
 	return xsp.screens[0]
 }
 
-func getScale(widthPx uint16, widthMm uint32) float32 {
-	env := os.Getenv("FYNE_SCALE")
-
-	if env != "" && env != "auto" {
-		scale, err := strconv.ParseFloat(env, 32)
-		if err == nil && scale != 0 {
-			return float32(scale)
-		}
-		fyne.LogError("Error reading scale", err)
-	}
-
-	if env != "auto" {
-		setting := fyne.CurrentApp().Settings().Scale()
-		switch setting {
-		case fyne.SettingsScaleAuto:
-			// fall through
-		case 0.0:
-			if env == "" {
-				return 1.0
-			}
-			// fall through
-		default:
-			return setting
-		}
-	}
+func getScale(widthPx, widthMm uint16) float32 {
 	dpi := float32(widthPx) / (float32(widthMm) / 25.4)
 	if dpi > 1000 || dpi < 10 {
 		dpi = 96
 	}
-	return float32(math.Round(float64(dpi)/144.0*10.0)) / 10.0
+	return float32(math.Round(float64(dpi)/96.0*10.0)) / 10.0
 }
 
-func (xsp *x11ScreensProvider) setupScreens(x *x11WM) {
-	err := randr.Init(x.x.Conn())
-	if err != nil {
-		fyne.LogError("Could not initialize randr", err)
-		xsp.setupSingleScreen(x)
-		return
-	}
-
-	root := xproto.Setup(x.x.Conn()).DefaultScreen(x.x.Conn()).Root
-	resources, err := randr.GetScreenResources(x.x.Conn(), root).Reply()
+func (xsp *x11ScreensProvider) setupScreens() {
+	root := xproto.Setup(xsp.x.x.Conn()).DefaultScreen(xsp.x.x.Conn()).Root
+	resources, err := randr.GetScreenResources(xsp.x.x.Conn(), root).Reply()
 	if err != nil || len(resources.Outputs) == 0 {
 		fyne.LogError("Could not get randr screen resources", err)
-		xsp.setupSingleScreen(x)
+		xsp.setupSingleScreen()
 		return
 	}
 
 	var primaryInfo *randr.GetOutputInfoReply
-	primary, err := randr.GetOutputPrimary(x.x.Conn(),
-		xproto.Setup(x.x.Conn()).DefaultScreen(x.x.Conn()).Root).Reply()
+	primary, err := randr.GetOutputPrimary(xsp.x.x.Conn(), root).Reply()
 	if err == nil {
-		primaryInfo, _ = randr.GetOutputInfo(x.x.Conn(), primary.Output, 0).Reply()
+		primaryInfo, _ = randr.GetOutputInfo(xsp.x.x.Conn(), primary.Output, 0).Reply()
 	}
 	primaryFound := false
-	var firstFoundMmWidth uint32
-	var firstFoundWidth uint16
-	i := 0
+	var tmpScreens []*desktop.Screen
 	for _, output := range resources.Outputs {
-		outputInfo, err := randr.GetOutputInfo(x.x.Conn(), output, 0).Reply()
+		outputInfo, err := randr.GetOutputInfo(xsp.x.x.Conn(), output, 0).Reply()
 		if err != nil {
 			fyne.LogError("Could not get randr output", err)
 			continue
 		}
 		if outputInfo.Crtc == 0 || outputInfo.Connection == randr.ConnectionDisconnected {
 			continue
 		}
-		crtcInfo, err := randr.GetCrtcInfo(x.x.Conn(), outputInfo.Crtc, 0).Reply()
+		crtcInfo, err := randr.GetCrtcInfo(xsp.x.x.Conn(), outputInfo.Crtc, 0).Reply()
 		if err != nil {
 			fyne.LogError("Could not get randr crtcs", err)
 			continue
 		}
-		if i == 0 {
-			firstFoundMmWidth = outputInfo.MmWidth
-			firstFoundWidth = crtcInfo.Width
+		insertIndex := -1
+		for i, screen := range tmpScreens {
```

Comment:
This chunk of new code adds to the complexity of this loop - perhaps it could be refactored out as "insertScreenInOrder" or something that describes the intent but removed the details from the loop